src/io.jl

```diff
@@ -85,32 +88,51 @@ function read_cellmap(habitat_file::String, is_res::Bool, ::Type{T}) where {T}
     gmap, rastermeta
 end
 
-function _ascii_grid_reader(T, file)
-    f = endswith(file, ".gz") ? GZip.open(file, "r") : open(file, "r")
-    rastermeta = _ascii_grid_read_header(file, f)
-    c = Matrix{T}(undef,0,0)
-    ss = 6
-    if rastermeta.nodata == -Inf
-        ss = 5
-    end
-    try
-        c = readdlm(f, T; skipstart = ss)
-    catch
-        seek(f, 0)
+function _grid_reader(T, file) # make a wrapper around read_raster to return rastermeta as well
+    #=if endswith(file, ".gz") ?
+        f = GZip.open(file, "r")
+        rastermeta = _ascii_grid_read_header(file, f)
+        c = Matrix{T}(undef,0,0)
+        ss = 6
+        if rastermeta.nodata == -Inf
+            ss = 5
+        end
         try
-            d = readdlm(f; skipstart = ss)
-            d = d[:, 1:end-1]
-            c = map(T, d)
+            c = readdlm(f, T; skipstart = ss)
         catch
-            error("Failed to read habitat map. There may be errors in your file.")
+            seek(f, 0)
+            try
+                d = readdlm(f; skipstart = ss)
+                d = d[:, 1:end-1]
+                c = map(T, d)
+            catch
+                error("Failed to read habitat map. There may be errors in your file.")
+            end
         end
-    end
-    map!(x -> x == rastermeta.nodata ? -9999. : x , c, c)
-    close(f)
-    c, rastermeta
+        map!(x -> x == rastermeta.nodata ? -9999. : x , c, c)
+        close(f)
+        return c, rastermeta
+    else=#
```

Comment:
This block can be deleted once okay'ed -- we will still need tests and potentially a solution for the Gzip stuff though if we want to keep it.

Reply:
deleted it. We'll need a different solution for GZip that was there originally anyway.